integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/GCPUtils.java

```diff
@@ -0,0 +1,251 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.util;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.common.io.CharStreams;
+import io.openlineage.client.Environment;
+import io.openlineage.spark.api.OpenLineageContext;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.http.Consts;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.apache.spark.SparkContext;
+import org.apache.spark.sql.execution.SparkPlan;
+import org.apache.spark.sql.execution.WholeStageCodegenExec;
+
+/** Util to extract values from GCP environment */
+public class GCPUtils {
+
+  private static final String PROJECT_ID_URI =
```

Comment:
Isn't there an endpoint to return all the required information. This would allow avoiding three separate http calls to get what is needed.

Reply:
Unfortunately, there isn't a single endpoint that returns everything. For example, `http://metadata.google.internal/computeMetadata/v1/instance/attributes` returns just a list of the keys available for that instance (`dataproc-region`, `dataproc-cluster-uuid` etc.)
Although these values are being cached, as far as I am aware.